controllers/factory/vmcluster.go

```diff
@@ -527,8 +527,8 @@ func makePodSpecForVMSelect(cr *v1beta1.VMCluster, c *config.BaseOperatorConf) (
 	}
 	if cr.Spec.ReplicationFactor != nil {
 		var dedupIsSet bool
-		for _, arg := range cr.Spec.VMSelect.ExtraArgs {
-			if strings.Contains(arg, "-dedup.minScrapeInterval") {
+		for arg, _ := range cr.Spec.VMSelect.ExtraArgs {
```

Comment:
```suggestion
		for arg := range cr.Spec.VMSelect.ExtraArgs {
```
 I suggest to change signature, in this case value can be omitted.